### ISSUE TITLE
Logs lost from tail plugin during unexpected power event (database, filesystem buffer enabled)

### ISSUE THREAD

**Issue by adriannawenz** [fluent/fluent-bit#6995]
## Bug Report

**Describe the bug**
TL;DR: Logs are lost from tail input plugin during an unexpected power event due to file system buffer corruption. Is there a way to not update the offset position until a file system chunk is fully written to the system? 

I work on an embedded linux system. Td-agent-bit is installed on the device. Sometimes the device will be unplugged, causing a power event. We are tailing the syslog file, use the DB option to keep our place, with full sync DB and full sync file system buffers. 

When the device is unplugged, it seems like the file system buffer chunks get corrupted. When the device is powered up again, it cannot process the corrupted chunks and begins reading the file from after those chunks. This means there are entries in syslog that never get processed by our pipeline. Is there a way to wait to update the DB until the file system chunks have been fully written to the system? 

Note: we would prefer to not have duplicated logs, as we are uploading directly to Google Cloud Logging via the Stackdriver plugin.

**To Reproduce**
This is the td-agent-bit.conf, with some inputs excluded since I don't think they relate to this problem. 

```
[SERVICE]
    flush        60
    daemon       Off
    log_level    info
    parsers_file parsers.conf
    plugins_file plugins.conf
    http_server  Off
    storage.metrics on
    storage.path /var/log/storage
    storage.sync full
    storage.delete_irrecoverable_chunks true

[INPUT]
    Name            tail
    Path            /var/log/syslog
    DB              /var/log/logs.db
    Read_from_Head  on
    DB.locking      true
    Skip_Long_Lines on
    Parser          syslog-rfc3164-custom
    storage.type    filesystem
    DB.sync         full

[OUTPUT]
    Name        stackdriver
    Match       *
    net.dns.resolver LEGACY
    Retry_Limit False
    net.dns.prefer_ipv4 true
    storage.total_limit_size 1G
```

**Example log message if applicable:**
```
GCP:
DEFAULT 2023-03-09T17:58:28Z E20230309 17:58:28.621646 5771 [my_file.cc](http://my_file.cc/):1] Application log
DEFAULT 2023-03-09T17:58:28Z E20230309 17:58:28.632647 5771 [my_file.cc](http://my_file.cc/):1] Application log

DEFAULT 2023-03-09T17:59:17Z Network interface NamePolicy= disabled on kernel command line, ignoring.
DEFAULT 2023-03-09T17:59:17Z Inserted module 'nvgpu'
DEFAULT 2023-03-09T17:59:17Z Couldn't write '1' to 'net/ipv4/tcp_syncookies', ignoring: No such file or directory
DEFAULT 2023-03-09T17:59:17Z Couldn't write '1' to 'kernel/yama/ptrace_scope', ignoring: No such file or directory
DEFAULT 2023-03-09T17:59:17Z Couldn't write 'fq_codel' to 'net/core/default_qdisc', ignoring: No such file or directory

Syslog:
Mar  9 17:58:57 48-b0-2d-1e-48-43 mender[7718]: time="2023-03-09T17:58:57Z" level=info msg="State transition: check-wait [Idle] -> inventory-update [Sync]"
Mar  9 17:58:58 48-b0-2d-1e-48-43 mender[7718]: time="2023-03-09T17:58:58Z" level=info msg="State transition: inventory-update [Sync] -> check-wait [Idle]"              
^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@Mar  9 17:59:17 48-b0-2d-1e-48-43 systemd-udevd[2341]: Network interface NamePolicy= disabled on kernel command line, ignoring.
Mar  9 17:59:17 48-b0-2d-1e-48-43 systemd-modules-load[2252]: Inserted module 'nvgpu'
Mar  9 17:59:17 48-b0-2d-1e-48-43 systemd-sysctl[2877]: Couldn't write '1' to 'net/ipv4/tcp_syncookies', ignoring: No such file or directory
```
**Steps to reproduce the problem:**

1. Enable td-agent-bit to tail a file, following the td-agent-bit.conf above.
2. Unplug the device.
3. Replug the device and observe that there are logs in syslog that are not in the output destination. When td-agent-bit starts up, it will say that filesystem chunks are corrupted.

**Expected behavior**
The DB won't be updated with a new offset until the new file system chunk has successfully been written to disk.

**Your Environment**
* Version used: v1.9.10
* Operating System and version: Jetson TX2, aarch64, Ubuntu 18.04.6 LTS
* Filters and plugins: tail, record_modifier, stackdriver. Others: http, cpu, disk, kmsg, mem, netif, proc, file (output)

**Additional context**
We want to use td-agent-bit to upload our logs into Google Cloud Logging. I'm open to other architectures, for example uploading syslog files to an intermediate server to process with td-agent-bit.

**Comment by agup006** [comment]
One strategy could be routing to another Fluent Bit server that buffers locally and then routes to Google Cloud Logging or elsewhere
